### ISSUE TITLE
Improvement: PWA for YouPHPTube

### ISSUE THREAD

**Issue by maxlinux2000** [WWBN/AVideo#2069]
Say it all ... apps are declining. Many people do not want to install new things to avoid having to run out of space on the smarthphone and not be able to download the video of the cute kittens from whatsapp.

In the case of the YouPHPTube app (very, very good by the way, the latest version) People have to go to playstore, install it, and then configure the server, and login ... too complicated for ordinary people.

I would pay for a plugin that adds a PWA to my server, so when a user signs up on a smartphone, they are asked if they want to add a link to the screen of their phone .... and there it is installed the PWA so that it has the logo of my site, the name that I want, and the user does not have to configure anything ... nor the login, Only the first time from firefox or chrome when create the account.

In short: I propose that you create a payment plugin to add a PWA app .... and make it cheap :)

**Comment by maxlinux2000** [comment]
I think that 50% of the work has already been done with the APIs that you have added to youphptube to manage the app. The later version of the app was a simple webview that pointed to the url of your demo. In short, it was a browser that could read only one address.
The version of the app that you now have instead read from the demo site a text file (a json I suppose) and use it as if it were a data base, loading the text in the boxes and downloading the images from the url provided in the json.

Well, a PWA will work the same. Everything is done in javascript, css and html5.

I have never developed a pwa, but I have read something and I think it is not difficult.
Is there documentation about the plugin APIs you've written? That maybe I help you with something :)

For example, can you make a call to get the list of videos? and User Login as you want it?

One thing ... PWAs do not work on iOS, therefore the app you have developed is still necessary. It doesn't work because Apple doesn't want to lose control of its playstore .... no comments!

Here a quick start in PWA develop from google
https://developers.google.com/web/fundamentals/codelabs/your-first-pwapp/

regards
MaX
